internal/auth/oidc/auth_method.go

```diff
@@ -0,0 +1,199 @@
+package oidc
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/boundary/internal/auth/oidc/store"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+	"github.com/hashicorp/go-kms-wrapping/structwrapping"
+	"google.golang.org/protobuf/proto"
+)
+
+// defaultAuthMethodTableName defines the default table name for an AuthMethod
+const defaultAuthMethodTableName = "auth_oidc_method"
+
+// AuthMethod contains an OIDC auth method configuration. It is owned
+// by a scope.  AuthMethods can have Accounts, AudClaims,
+// CallbackUrls, Certificates, SigningAlgs.  AuthMethods also have one State at
+// any given time which determines it's behavior for many its operations.
+type AuthMethod struct {
+	*store.AuthMethod
+	tableName string
+}
+
+// NewAuthMethod creates a new in memory AuthMethod assigned to scopeId.
+// WithMaxAge, WithName and WithDescription are the only valid options. All
+// other options are ignored.
+//
+// State equals the state of the OIDC auth method.  State is not a supported
+// parameter when creating new AuthMethod's since it must be Inactive for all
+// new AuthMethods.
+//
+// DiscoveryUrl equals a URL where the OIDC provider's configuration can be
+// retrieved. The OIDC Discovery Specification requires the URL end in
+// /.well-known/openid-configuration. However, Boundary will strip off
+// anything beyond scheme, host and port
+//
+// ClientId equals an OAuth 2.0 Client Identifier valid at the Authorization
+// Server.
+//
+// ClientSecret equals the client's secret which will be encrypted when stored
+// in the database and an hmac representation will also be stored when ever the
+// secret changes.  The secret is not returned via the API, the hmac is returned
+// so callers can determine if it's been updated.
+//
+// MaxAge equals the Maximum Authentication Age. Specifies the allowable elapsed
+// time in seconds since the last time the End-User was actively authenticated
+// by the OP. If the elapsed time is greater than this value, the OP MUST
+// attempt to actively re-authenticate the End-User. A value -1 basically
+// forces the IdP to re-authenticate the End-User.  Zero is not a valid value.
+//
+// See: https://openid.net/specs/openid-connect-core-1_0.html
+func NewAuthMethod(scopeId string, discoveryUrl *url.URL, clientId string, clientSecret ClientSecret, opt ...Option) (*AuthMethod, error) {
+	const op = "oidc.NewAuthMethod"
+
+	var u string
```

Comment:
Could this var name be updated so the code that creates it doesn't have to be read? 
On the other hand, you only have to look down a couple lines and figure it out. 🤷🏻  I dunno

Reply:
if they were separated a bit more in the code, then maybe.